File: frontend/src/types/material.ts
/** 库存状态 */
export type StockStatus = 'normal' | 'low' | 'high' | 'expired'

/** 物料状态 */
export type MaterialStatus = 'active' | 'inactive'

/** 物料信息 */
export interface Material {
  id: number
  materialCode: string
  materialName: string
  materialSpec: string
  unit: string
  categoryName: string
  shelfLifeDays: number
  nearExpiryDays: number
  warningDays: number
  minStock: number
  maxStock: number
  currentStock: number
  stockStatus: StockStatus
  storageRequire: string
  imageUrl: string
  status: MaterialStatus
  remark: string
  foodItemId?: number | null
  foodCode?: string | null
  foodName?: string | null
  calories?: number | null
  protein?: number | null
  carbohydrate?: number | null
  fat?: number | null
  sodium?: number | null
  fiber?: number | null
  vitaminA?: number | null
  vitaminB1?: number | null
  vitaminB2?: number | null
  vitaminC?: number | null
  vitaminE?: number | null
  calcium?: number | null
  iron?: number | null
  zinc?: number | null
  nutritionSourceType?: string | null
  nutritionSyncedAt?: string | null
  sourceVersion?: string | null
  expiryDate?: string
  createdAt: string
  updatedAt: string
}

/** 物料表单 */
export interface MaterialForm {
  materialCode: string
  materialName: string
  materialSpec: string
  unit: string
  categoryName: string
  shelfLifeDays: number
  nearExpiryDays: number
  warningDays: number
  minStock: number
  maxStock: number
  storageRequire: string
  imageUrl: string
  remark: string
}

/** 物料查询参数 */
export interface MaterialQuery {
  pageNum: number
  pageSize: number
  materialName?: string
  materialCode?: string
  categoryName?: string
  stockStatus?: StockStatus
  status?: MaterialStatus
}

/** 物料统计数据 */
export interface MaterialStatistics {
  total: number
  activeCount: number
  inactiveCount: number
  incompleteCount: number
}

/** 物料导入结果 */
export interface MaterialImportResult {
  total: number
  successCount: number
  failCount: number
  hasErrors: boolean
  errorFileUrl: string | null
}

export interface FoodCategory {
  id: number
  parentId?: number | null
  categoryCode: string
  categoryName: string
  categoryLevel: number
  sortOrder: number
  sourceFile?: string
  status: string
}

export interface FoodItem {
  id: number
  foodCode: string
  foodName: string
  categoryLevel1Id?: number | null
  categoryLevel2Id?: number | null
  edibleRatio?: number | null
  energyKcal?: number | null
  protein?: number | null
  fat?: number | null
  carbohydrate?: number | null
  dietaryFiber?: number | null
  sodium?: number | null
  vitaminA?: number | null
  vitaminB1?: number | null
  vitaminB2?: number | null
  vitaminC?: number | null
  vitaminE?: number | null
  calcium?: number | null
  iron?: number | null
  zinc?: number | null
  sourceFile?: string
  sourceVersion?: string
  status: string
}

export interface FoodImportResult {
  fileCount: number
  categoryCount: number
  itemCount: number
  sourceVersion: string
}

export interface MaterialFoodMapping {
  id: number
  materialId: number
  materialCode?: string
  materialName?: string
  foodItemId: number
  foodCode?: string
  foodName?: string
  matchStatus: string
  remark?: string
  confirmedAt?: string
}

export interface MaterialFoodMappingQuery {
  pageNum: number
  pageSize: number
  materialId?: number
  materialName?: string
  matchStatus?: string
}

export interface MaterialFoodMappingForm {
  materialId: number
  foodItemId: number
  matchStatus?: string
  remark?: string
}

export interface NutritionSyncResult {
  materialId: number
  foodItemId: number
  materialName: string
  foodName: string
  nutritionSourceType: string
}

File: backend/wms-service/src/main/java/com/xykj/wms/entity/Material.java
package com.xykj.wms.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 物料主数据实体
 * 对应数据库表: wms_material
 */
@Data
@TableName("wms_material")
public class Material implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 物料ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 物料编码
     */
    private String materialCode;

    /**
     * 物料名称
     */
    private String materialName;

    /**
     * 物料分类
     */
    private String materialCategory;

    /**
     * 关联标准食品ID
     */
    private Long foodItemId;

    /**
     * 基本单位
     */
    private String unit;

    /**
     * 规格说明
     */
    private String spec;

    /**
     * 保质期（天）
     */
    private Integer shelfLifeDays;

    /**
     * 最低库存预警量
     */
    private BigDecimal minStock;

    /**
     * 最高库存量
     */
    private BigDecimal maxStock;

    /**
     * 临期提醒天数（到期前N天提醒）
     */
    private Integer nearExpiryDays;

    /**
     * 预警天数（采购预警）
     */
    private Integer warningDays;

    /**
     * 存储条件说明
     */
    private String storageConditions;

    /**
     * 存储类型: normal=常温, cold=冷藏, freeze=冷冻
     */
    private String storageType;

    /**
     * 物料图片URL
     */
    private String imageUrl;

    /**
     * 状态: active=启用, inactive=停用
     */
    private String status;

    /**
     * 备注
     */
    private String remark;

    private BigDecimal calories;
    private BigDecimal protein;
    private BigDecimal carbohydrate;
    private BigDecimal fat;
    private BigDecimal sodium;
    private BigDecimal fiber;
    private BigDecimal vitaminA;
    private BigDecimal vitaminB1;
    private BigDecimal vitaminB2;
    private BigDecimal vitaminC;
    private BigDecimal vitaminE;
    private BigDecimal calcium;
    private BigDecimal iron;
    private BigDecimal zinc;
    private String nutritionSourceType;
    private Long nutritionSourceRefId;
    private LocalDateTime nutritionSyncedAt;

    /**
     * 所属组织ID
     */
    private Long orgId;

    /**
     * 租户ID
     */
    private Long tenantId;

    /**
     * 创建人ID
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新人ID
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 逻辑删除: 0=未删除, 1=已删除
     */
    @TableLogic
    private Integer deleted;
}
